### ISSUE TITLE
Sometimes the shutdown hangs

### ISSUE THREAD

**Issue by genesiscz** [beyondcode/laravel-websockets#729]
Trying to execute the worker, I am getting

<img width="322" alt="image" src="https://user-images.githubusercontent.com/1512353/113021852-49838a00-9184-11eb-8ccb-70b4561a3778.png">

I have to kill the process after

**Comment by rennokki** [comment]
Linux/Windows?

**Comment by genesiscz** [comment]
Debian

**Comment by genesiscz** [comment]
![image](https://user-images.githubusercontent.com/1512353/113626857-f79aa280-9662-11eb-86b6-6814afdf96fb.png)
pretty weird, it starts to use so much memory on quitting ... I have to kill the process with -9

**Comment by rennokki** [comment]
Can you specify how your `websocket` configuration looks like? I'm trying to find out which drivers or connectors you use.

**Comment by genesiscz** [comment]
Pretty much unchanged I think, not 100% sure tho. SSL are 100% changed (btw they silently fail when expired), not much more


```
<?php

use BeyondCode\LaravelWebSockets\Dashboard\Http\Middleware\Authorize;
return [
	'dashboard' => [

		'port' => env('LARAVEL_WEBSOCKETS_PORT', 6001),

		'path' => 'laravel-websockets',

		'middleware' => [
			'web',
			\BeyondCode\LaravelWebSockets\Dashboard\Http\Middleware\Authorize::class,
		],

	],

	'managers' => [

		/*
		|--------------------------------------------------------------------------
		| Application Manager
		|--------------------------------------------------------------------------
		|
		| An Application manager determines how your websocket server allows
		| the use of the TCP protocol based on, for example, a list of allowed
		| applications.
		| By default, it uses the defined array in the config file, but you can
		| anytime implement the same interface as the class and add your own
		| custom method to retrieve the apps.
		|
		*/

		'app' => \BeyondCode\LaravelWebSockets\Apps\ConfigAppManager::class,

	],

	/*
	|--------------------------------------------------------------------------
	| Applications Repository
	|--------------------------------------------------------------------------
	|
	| By default, the only allowed app is the one you define with
	| your PUSHER_* variables from .env.
	| You can configure to use multiple apps if you need to, or use
	| a custom App Manager that will handle the apps from a database, per se.
	|
	| You can apply multiple settings, like the maximum capacity, enable
	| client-to-client messages or statistics.
	|
	*/

	'apps' => [
		[
			'id' => env('PUSHER_APP_ID'),
			'name' => env('APP_NAME'),
			'host' => env('PUSHER_APP_HOST'),
			'key' => env('PUSHER_APP_KEY'),
			'secret' => env('PUSHER_APP_SECRET'),
			'path' => env('PUSHER_APP_PATH'),
			'capacity' => null,
			'enable_client_messages' => true,
			'enable_statistics' => true,
			'allowed_origins' => [
				//
			],
		],
	],

	/*
	|--------------------------------------------------------------------------
	| Broadcasting Replication PubSub
	|--------------------------------------------------------------------------
	|
	| You can enable replication to publish and subscribe to
	| messages across the driver.
	|
	| By default, it is set to 'local', but you can configure it to use drivers
	| like Redis to ensure connection between multiple instances of
	| WebSocket servers. Just set the driver to 'redis' to enable the PubSub using Redis.
	|
	*/

	'replication' => [

		'mode' => env('WEBSOCKETS_REPLICATION_MODE', 'local'),

		'modes' => [

			/*
			|--------------------------------------------------------------------------
			| Local Replication
			|--------------------------------------------------------------------------
			|
			| Local replication is actually a null replicator, meaning that it
			| is the default behaviour of storing the connections into an array.
			|
			*/

			'local' => [

				/*
				|--------------------------------------------------------------------------
				| Channel Manager
				|--------------------------------------------------------------------------
				|
				| The channel manager is responsible for storing, tracking and retrieving
				| the channels as long as their members and connections.
				|
				*/

				'channel_manager' => \BeyondCode\LaravelWebSockets\ChannelManagers\LocalChannelManager::class,

				/*
				|--------------------------------------------------------------------------
				| Statistics Collector
				|--------------------------------------------------------------------------
				|
				| The Statistics Collector will, by default, handle the incoming statistics,
				| storing them until they will become dumped into another database, usually
				| a MySQL database or a time-series database.
				|
				*/

				'collector' => \BeyondCode\LaravelWebSockets\Statistics\Collectors\MemoryCollector::class,

			],

			'redis' => [

				'connection' => env('WEBSOCKETS_REDIS_REPLICATION_CONNECTION', 'default'),

				/*
				|--------------------------------------------------------------------------
				| Channel Manager
				|--------------------------------------------------------------------------
				|
				| The channel manager is responsible for storing, tracking and retrieving
				| the channels as long as their members and connections.
				|
				*/

				'channel_manager' => \BeyondCode\LaravelWebSockets\ChannelManagers\RedisChannelManager::class,

				/*
				|--------------------------------------------------------------------------
				| Statistics Collector
				|--------------------------------------------------------------------------
				|
				| The Statistics Collector will, by default, handle the incoming statistics,
				| storing them until they will become dumped into another database, usually
				| a MySQL database or a time-series database.
				|
				*/

				'collector' => \BeyondCode\LaravelWebSockets\Statistics\Collectors\RedisCollector::class,

			],

		],

	],

	'statistics' => [

		/*
		|--------------------------------------------------------------------------
		| Statistics Store
		|--------------------------------------------------------------------------
		|
		| The Statistics Store is the place where all the temporary stats will
		| be dumped. This is a much reliable store and will be used to display
		| graphs or handle it later on your app.
		|
		*/

		'store' => \BeyondCode\LaravelWebSockets\Statistics\Stores\DatabaseStore::class,

		/*
		|--------------------------------------------------------------------------
		| Statistics Interval Period
		|--------------------------------------------------------------------------
		|
		| Here you can specify the interval in seconds at which
		| statistics should be logged.
		|
		*/

		'interval_in_seconds' => 60,

		/*
		|--------------------------------------------------------------------------
		| Statistics Deletion Period
		|--------------------------------------------------------------------------
		|
		| When the clean-command is executed, all recorded statistics older than
		| the number of days specified here will be deleted.
		|
		*/

		'delete_statistics_older_than_days' => 60,

	],

	/*
	|--------------------------------------------------------------------------
	| Maximum Request Size
	|--------------------------------------------------------------------------
	|
	| The maximum request size in kilobytes that is allowed for
	| an incoming WebSocket request.
	|
	*/

	'max_request_size_in_kb' => 250,

	/*
	|--------------------------------------------------------------------------
	| SSL Configuration
	|--------------------------------------------------------------------------
	|
	| By default, the configuration allows only on HTTP. For SSL, you need
	| to set up the the certificate, the key, and optionally, the passphrase
	| for the private key.
	| You will need to restart the server for the settings to take place.
	|
	*/

	'ssl' => [
		'capath' => env('LARAVEL_WEBSOCKETS_SSL_CA', null),
		/*
		* Path to local certificate file on filesystem. It must be a PEM encoded file which
		* contains your certificate and private key. It can optionally contain the
		* certificate chain of issuers. The private key also may be contained
		* in a separate file specified by local_pk.
		*/
	
		'local_cert' => __DIR__ . "/../certs/".HALL_ID.".crt",//env('LARAVEL_WEBSOCKETS_SSL_LOCAL_CERT', __DIR__."/../certs/speeder.pem"),

		/*
		 * Path to local private key file on filesystem in case of separate files for
		 * certificate (local_cert) and private key.
		 */
		'local_pk' => __DIR__ . "/../certs/".HALL_ID.".pem",//env('LARAVEL_WEBSOCKETS_SSL_LOCAL_PK', null),

		'passphrase' => env('LARAVEL_WEBSOCKETS_SSL_PASSPHRASE', null),

		'verify_peer' => env('APP_ENV') === 'production',

		'allow_self_signed' => env('APP_ENV') !== 'production',

	],

	/*
	|--------------------------------------------------------------------------
	| Route Handlers
	|--------------------------------------------------------------------------
	|
	| Here you can specify the route handlers that will take over
	| the incoming/outgoing websocket connections. You can extend the
	| original class and implement your own logic, alongside
	| with the existing logic.
	|
	*/

	'handlers' => [

		'websocket' => \BeyondCode\LaravelWebSockets\Server\WebSocketHandler::class,

		'health' => \BeyondCode\LaravelWebSockets\Server\HealthHandler::class,

		'trigger_event' => \BeyondCode\LaravelWebSockets\API\TriggerEvent::class,

		'fetch_channels' => \BeyondCode\LaravelWebSockets\API\FetchChannels::class,

		'fetch_channel' => \BeyondCode\LaravelWebSockets\API\FetchChannel::class,

		'fetch_users' => \BeyondCode\LaravelWebSockets\API\FetchUsers::class,

	],

	/*
	|--------------------------------------------------------------------------
	| Promise Resolver
	|--------------------------------------------------------------------------
	|
	| The promise resolver is a class that takes a input value and is
	| able to make sure the PHP code runs async by using ->then(). You can
	| use your own Promise Resolver. This is usually changed when you want to
	| intercept values by the promises throughout the app, like in testing
	| to switch from async to sync.
	|
	*/

	'promise_resolver' => \React\Promise\FulfilledPromise::class,

];

```

**Comment by simonbuehler** [comment]
pls provide a profile dump via https://xdebug.org/docs/profiler#starting

**Comment by genesiscz** [comment]
> pls provide a profile dump via https://xdebug.org/docs/profiler#starting

Is that even manageable with long-running script like this? I am not sure after what time or events does that happen, so not sure if I am able to handle such a big amount of profiling

**Comment by simonbuehler** [comment]
worth a try, you can cancel it when a significant memory increase has started.

syntax for cli: `php -d xdebug.profiler_enable=On -d xdebug.xxxx ... php artisan ...`

**Comment by genesiscz** [comment]
I managed to fix the problem! I was quite foolish to set `storage.database.chunk` to 100 000 in telescope.php config. Because Telescope writes its entries at the end of the command, it tried to write days of queries/events/models/whatever at once (at least 100k entries in one INSERT clause). I set chunk to 100 000 to prune the database pretty fast, but the INSERT uses the same config value.

Maybe it would make a sense to somehow call `Telescope::store(app(\Laravel\Telescope\Contracts\EntriesRepository::class))` somewhere to release the pressure?
